tests/controls/treectrltest.cpp

```diff
@@ -111,134 +84,103 @@ class TreeCtrlTestCase : public CppUnit::TestCase
     wxDECLARE_NO_COPY_CLASS(TreeCtrlTestCase);
 };
 
-// register in the unnamed registry so that these tests are run by default
-CPPUNIT_TEST_SUITE_REGISTRATION( TreeCtrlTestCase );
-
-// also include in its own registry so that these tests can be run alone
-CPPUNIT_TEST_SUITE_NAMED_REGISTRATION( TreeCtrlTestCase, "TreeCtrlTestCase" );
-
 // ----------------------------------------------------------------------------
-// test initialization
+// the tests themselves
 // ----------------------------------------------------------------------------
 
-bool TreeCtrlTestCase::ms_multiSelect = false;
-bool TreeCtrlTestCase::ms_hiddenRoot = false;
-
-void TreeCtrlTestCase::setUp()
+TEST_CASE_METHOD(TreeCtrlTestCase, "wxTreeCtrl::HasChildren", "[treectrl]")
 {
-    m_tree = new wxTreeCtrl(wxTheApp->GetTopWindow(),
-                            wxID_ANY,
-                            wxDefaultPosition,
-                            wxSize(400, 200),
-                            wxTR_DEFAULT_STYLE | wxTR_EDIT_LABELS);
-
-    if ( ms_multiSelect )
-        m_tree->ToggleWindowStyle(wxTR_MULTIPLE);
-
-    if ( ms_hiddenRoot )
-        m_tree->ToggleWindowStyle(wxTR_HIDE_ROOT); // actually set it
-
-    m_root = m_tree->AddRoot("root");
-    m_child1 = m_tree->AppendItem(m_root, "child1");
-    m_child2 = m_tree->AppendItem(m_root, "child2");
-    m_grandchild = m_tree->AppendItem(m_child1, "grandchild");
-
-    m_tree->SetSize(400, 200);
-    m_tree->ExpandAll();
-    m_tree->Refresh();
-    m_tree->Update();
-}
-
-void TreeCtrlTestCase::tearDown()
-{
-    delete m_tree;
-    m_tree = nullptr;
+#ifdef __WXMSW__
+    // Toggling the wxTR_HIDE_ROOT window style has no effect under wxMSW if
```

Comment:
It would be nice to mention this in `wxTR_HIDE_ROOT` docs.

Reply:
Oops! I forgot to update the documentation. Please could you do this when merging?

TIA!